packages/shared-react/src/api/queries/transactionQueries.ts

```diff
@@ -164,21 +165,36 @@ export const walletTransactionsInfiniteQuery = ({
           } as PageParam)
   })
 
-interface TransactionQueryProps extends SkipProp {
+interface PendingTransactionQueryProps extends SkipProp {
   txHash: string
   networkId?: number
 }
 
-export const confirmedTransactionQuery = ({ txHash, networkId, skip }: TransactionQueryProps) =>
+interface ConfirmedTransactionQueryProps extends PendingTransactionQueryProps {
+  addressHashes: AddressHash[]
+}
+
+export const confirmedTransactionQuery = ({ txHash, addressHashes, networkId, skip }: ConfirmedTransactionQueryProps) =>
   queryOptions({
     queryKey: ['transaction', 'confirmed', txHash],
     // When the user navigates away from the transaction details modal for 5 minutes or when a sent tx confirms the
     // cached data will be deleted.
     ...getQueryConfig({ staleTime: Infinity, gcTime: FIVE_MINUTES_MS, networkId }),
-    queryFn: !skip ? () => throttledClient.explorer.transactions.getTransactionsTransactionHash(txHash) : skipToken
+    queryFn: !skip ? () => throttledClient.explorer.transactions.getTransactionsTransactionHash(txHash) : skipToken,
+    placeholderData: () => {
+      const paginatedTxs = queryClient.getQueryData(
+        walletTransactionsInfiniteQuery({ addressHashes, networkId }).queryKey
+      )
+
+      const tx = paginatedTxs?.pages
+        .flatMap(({ pageTransactions }) => pageTransactions)
+        .find((tx) => tx.hash === txHash)
+
+      return tx ? ({ ...tx, type: 'confirmed' } as AcceptedTransaction) : undefined
+    }
```

Comment:
Prefills the transaction query with data we might already have because of the transactions infinite query. Improves the UX when opening the tx details modal on DW

Reply:
Really nice.